purchase_order_reorder_lines/purchase.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+#
+#    Author: Alexandre Fayolle
+#    Copyright 2013 Camptocamp SA
+#    Author: Damien Crier
+#    Copyright 2015 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+from openerp import models, api, fields
+from openerp.tools.safe_eval import safe_eval
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+    _order = 'order_id desc, sequence, id'
+
+    sequence = fields.Integer(help="Gives the sequence of this line when "
+                                   "displaying the purchase order.")
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.model
+    def _prepare_inv_line(self, account_id, order_line):
+        res = super(PurchaseOrder, self)._prepare_inv_line(
+            account_id,
+            order_line,
+            )
+        res['sequence'] = order_line.sequence
+        return res
+
+    @api.model
+    def _prepare_order_line_move(self, order, order_line,
+                                 picking_id, group_id):
+        res = super(PurchaseOrder, self)._prepare_order_line_move(
+            order,
+            order_line,
+            picking_id,
+            group_id)
+        if res:
+            res[0]['sequence'] = order_line.sequence
+        return res
+
+    @api.depends('order_line')
+    def _get_max_line_sequence(self):
```

Comment:
Can you extend with a little docstring why is this method for (functionally speaking)?